// Copyright (c) 2025, Alliance for Open Media. All rights reserved
//
// This source code is subject to the terms of the BSD 3-Clause Clear License
// and the Alliance for Open Media Patent License 1.0. If the BSD 3-Clause Clear
// License was not distributed with this source code in the LICENSE file, you
// can obtain it at www.aomedia.org/license/software-license/bsd-3-c-c. If the
// Alliance for Open Media Patent License 1.0 was not distributed with this
// source code in the PATENTS file, you can obtain it at
// www.aomedia.org/license/patent.

syntax = "proto2";

package iamf_tools_cli_proto;

import "iamf/cli/proto/output_audio_format.proto";

// Controls for the encoder behavior.
message EncoderControlMetadata {
  // If true [default]: Each Mix Presentation OBU will get an extra
  //                    `MixPresentationTag` with metadata describing the build
  //                    information.
  //  If false: The encoder omit adding an extra build information tag.
  optional bool add_build_information_tag = 1 [default = true];

  // Controls the output audio format for any rendered files. The encoder may
  // produce rendered files as a side effect of the encoding process. The files
  // are useful for debugging and as an approximation of what a decoder would
  // generate when playing back the IAMF file.
  optional OutputAudioFormat output_rendered_file_format = 2
      [default = OUTPUT_FORMAT_NONE];
}
